From d60f57f53682fc0966a454b42ef29380b6e0cf08 Mon Sep 17 00:00:00 2001
From: HAY Steve <steve.hay@hexagon.com>
Date: Wed, 5 Jul 2023 14:35:24 +0100
Subject: [PATCH 06/12] Check for 'libexpat' rather than 'expat' on Windows

With Expat built (in its default configuration) and installed into
C:\Programs\expat the following command should work but doesn't unless we
check for 'libexpat' rather than 'expat':
perl Makefile.PL EXPATLIBPATH=C:\Programs\expat\lib EXPATINCPATH=C:\Programs\expat\include

(The (DLL import) libray is C:\Programs\expat\lib\libexpat.lib. The DLL
itself is C:\Programs\expat\bin\libexpat.dll.)

Signed-off-by: Kevin Cui <bh@bugs.cc>
---
 Makefile.PL | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile.PL b/Makefile.PL
index 505d1df..5d46e8d 100644
--- a/Makefile.PL
+++ b/Makefile.PL
@@ -30,7 +30,7 @@ foreach (@ARGV) {
 
 unless (
     check_lib(    # fill in what you prompted the user for here
-        lib     => [qw(expat)],
+        lib     => [$^O eq 'MSWin32' ? qw(libexpat) : qw(expat)],
         header  => ['expat.h'],
         incpath => $expat_incpath,
         ( $expat_libpath ? ( libpath => $expat_libpath ) : () ),
-- 
2.43.0

